modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/autocomplete/Autocomplete.js

```diff
@@ -125,232 +114,91 @@ function Autocomplete({onChange, onItemsUpdated, onValueUpdated, ...props}) {
 						return jsonResponse.items;
 					});
 
-					updateTotalCount(jsonResponse.totalCount);
-					updateLastPage(jsonResponse.lastPage);
+					setTotalCount(jsonResponse.totalCount);
+					setLastPage(jsonResponse.lastPage);
 					setLoading(false);
 
 					if (!query) {
 						return;
 					}
+
 					const found = jsonResponse.items.find(
-						(item) =>
-							getValueFromItem(item, props.itemsLabel) === query
+						(item) => getValueFromItem(item, itemsLabel) === query
 					);
+
 					if (found) {
-						updateSelectedItem(found);
+						setSelectedItem(found);
 					}
 				})
 				.catch(() => {
 					showErrorNotification();
 					setLoading(false);
 				});
 		}
-	}, [
-		debouncedGetItems,
-		initialised,
-		isMounted,
-		query,
-		page,
-		pageSize,
-		props.disabled,
-		props.infiniteScrollMode,
-		props.apiUrl,
-		props.itemsLabel,
-		props.showErrorNotification,
-	]);
-
-	useEffect(() => {
-		if (onItemsUpdated) {
-			onItemsUpdated(items);
-		}
-	}, [items, onItemsUpdated]);
-
-	useEffect(() => {
-		function handleClick(event) {
-			if (
-				node.current.contains(event.target) ||
-				(dropdownNode.current &&
-					dropdownNode.current.contains(event.target))
-			) {
-				return;
-			}
-
-			setActive(false);
-		}
-		if (active) {
-			document.addEventListener('mousedown', handleClick);
-		}
-
-		return () => {
-			document.removeEventListener('mousedown', handleClick);
-		};
-	}, [active]);
-
-	let results;
+	}, fetchDataDebounce);
+
+	const itemsWrapperRenderer = ({items, labelKey, onSelect, valueKey}) => (
+		<InfiniteScroller
+			onBottomTouched={() => {
+				if (!loading) {
+					setPage((currentPage) =>
+						currentPage < lastPage ? currentPage + 1 : currentPage
+					);
+				}
+			}}
+			scrollCompleted={!items || items.length === totalCount}
+		>
+			{CustomView ? (
+				<CustomView
+					items={items}
+					lastPage={lastPage}
+					loading={loading}
+					page={page}
+					pageSize={internalPageSize}
+					setPage={setPage}
+					setPageSize={setInternalPageSize}
+					setSelectedItem={setSelectedItem}
```

Comment:
I know you're just moving this component to another module/folder but it think it can be changed to a callback or a reducer(depending the depth)

We can see many traverse of state setters everywhere in this code :/